### PR TITLE
Google translate refactoring, based on select form element,

### LINKED ISSUE #1328: Language translation on mobile issues
Across all themes, the language translation component is not displaying properly on mobile. See examples below for  Lily, but the same issue is on Rose.

Further, note that when Lily has the language translation block open, the navigation can't be closed, which also needs to be fixed.

Done when:
1) Language block displays properly on mobile and all languages can be selected
2) The menu can be closed on Lily even if a language hasn't been selected

file

### PR DESCRIPTION
**Pull request by anpolimus**
Original Issue, this PR is going to fix: https://github.com/ymcatwincities/openy/issues/1328

At this PR I've rebuilt google translator block according to select approach. This approach fixes issues at mobile devices and make translation widget more stable and easy to use.

Initial approach from @podarok had upgrade path that were deleting language menu item from mobile menu. At my approach I'm using this menu item as entry point for displaying copied content from header to mobile menu section.

Screenshots of how this widget looks like at different themes:
*Carnation*
image
image
*LiLy*
image
image
*Rose*
image
image

## Steps for review

- [ ] .
- [ ] Please provide steps for review here.
- [ ] Please provide steps for review here.
- [ ] Please provide steps for review here.

## General checks
- [ ] All coding styles are fulfilled and there are no any issues reported by CodeSniffer CI. <br/>CI code sniffer errors
- [ ] All tests are running and there are no failed tests reported by CI. <br/>Behat test results
- [ ] [Documentation](https://github.com/ymcatwincities/openy/tree/8.x-1.x/docs) has been updated according to PR changes.
- [ ] [Steps for review](https://github.com/ymcatwincities/openy/pull/94#issue-204580200) have been provided according to PR changes. <br/>Steps for review
- [ ] Make sure you've provided all necessary hook\_update\_N to [support upgrade path](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Upgrade%20path.md).
- [ ] Make sure your git email is associated with account on drupal.org, otherwise you won't get commits there. <br/>drupal.org email
- [ ] If you would like to get credits on drupal.org, [check documentation](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Contributing.md#drupalorg-credits).

Thank you for your contribution!

### CODE REVIEW

**Comment by bjwilkins** [comment]
Can one of the admins verify this patch? Use "o+k to test" or ''t+est this please" for manual build execution.

**Comment by podarok** [comment]
ok to test

File: modules/custom/openy_gtranslate/templates/openy-gtranslate.html.twig [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
+            {pageLanguage: drupalSettings.path.currentLanguage, layout: google.translate.TranslateElement.InlineLayout.VERTICAL},
+            'google_translate_element');
+        if(typeof(document.querySelector) == 'function') {
+            document.querySelector('.goog-logo-link').setAttribute('style', 'display: none');
+            document.querySelector('.goog-te-gadget').setAttribute('style', 'font-size: 0');
+        }
+    }
 </script>
-<script async type="text/javascript" src="//translate.google.com/translate_a/element.js?cb=googleTranslateElementInit"></script>
+<script type="text/javascript" src="//translate.google.com/translate_a/element.js?cb=googleTranslateElementInit"></script>
```
Comment (podarok) [inline modules/custom/openy_gtranslate/templates/openy-gtranslate.html.twig]: let's keep async for PageSpeed optimization

**Review by podarok — CHANGES_REQUESTED** [review]
Let's keep PageSpeed optimizations in place

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation - Rose theme | http://openy.cibox.tools/build1635 |
| Fresh OpenY installation - Lily theme | http://lily.openy.cibox.tools/build1635 |
| Upgraded(upgrade path) installation | http://upgrade.openy.cibox.tools/build1635 |
| Installation process | http://install.openy.cibox.tools/build1635/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/1635/

File: modules/custom/openy_gtranslate/templates/openy-gtranslate.html.twig (hunk 1 above)
Comment (anpolimus) [reply modules/custom/openy_gtranslate/templates/openy-gtranslate.html.twig]: @podarok, fixed.

**Comment by gundevel** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation - Rose theme | http://openy.cibox.tools/build1636 |
| Fresh OpenY installation - Lily theme | http://lily.openy.cibox.tools/build1636 |
| Upgraded(upgrade path) installation | http://upgrade.openy.cibox.tools/build1636 |
| Installation process | http://install.openy.cibox.tools/build1636/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/1636/

**Comment by gundevel** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build1636/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build1636/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build1636/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build1636/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build1636/blogWCAG2AAhtmlcs.txt

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation - Rose theme | http://openy.cibox.tools/build1641 |
| Fresh OpenY installation - Lily theme | http://lily.openy.cibox.tools/build1641 |
| Upgraded(upgrade path) installation | http://upgrade.openy.cibox.tools/build1641 |
| Installation process | http://install.openy.cibox.tools/build1641/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/1641/

**Pull request merged**